src/span.h

```diff
@@ -18,6 +18,16 @@
 #define ASSERT_IF_DEBUG(x)
 #endif
 
+#if defined(__clang__)
+#if __has_attribute(lifetimebound)
+#define SPAN_ATTR_LIFETIMEBOUND [[clang::lifetimebound]]
```

Comment:
```suggestion
#define ATTR_LIFETIMEBOUND [[clang::lifetimebound]]
```

Can this be named a bit more generic, so that it can be used in other places such as #19426 ?

Reply:
I namespaced it to avoid colliding with a more general define. As ```span.h``` is designed to be an abstraction, I'd really prefer to keep it an island and not require that it include another header.

Since @sipa was quick to point out a few places where this could already be useful, how about going with @practicalswift's [idea](https://github.com/bitcoin/bitcoin/pull/19387#issuecomment-650752959) of adding a second, more generic define in ```attributes.h``` ?

Reply:
@theuni Not sure how to weigh code duplication vs. a dependency on attributes.h. I'm fine with either.